packages/lib-sourcify/test/Verification/Verification.spec.ts

```diff
@@ -0,0 +1,1213 @@
+                offset: 4148,
+                id: '1',
+              },
+              {
+                type: 'replace',
+                reason: 'cborAuxdata',
+                offset: 2775,
+                id: '2',
+              },
+              {
+                type: 'replace',
+                reason: 'cborAuxdata',
+                offset: 4095,
+                id: '3',
+              },
+            ],
+            values: {
+              cborAuxdata: {
+                '1': '0xa2646970667358221220fe338e4778c1623b5865cd4121849802c8ed68e688def4d95b606f2f02ec563e64736f6c63430008090033',
+                '2': '0xa2646970667358221220bc654cadfb13b9ef229b6a2db4424f95dc4c52e3ae9b60648aa276f8eb0b3f8464736f6c63430008090033',
+                '3': '0xa2646970667358221220eb4312065a8c0fb940ef11ef5853554a447a5325095ee0f8fbbbbfc43dbb1b7464736f6c63430008090033',
+              },
+            },
+          },
+        },
+      });
+    });
+
+    it('should verify a library with call protection and add the transformation', async () => {
+      const contractFolderPath = path.join(
+        __dirname,
+        '..',
+        'sources',
+        'CallProtectionForLibraries',
+      );
+      const { contractAddress } = await deployFromAbiAndBytecode(
+        signer,
+        contractFolderPath,
+      );
+
+      const compilation = await getCompilationFromMetadata(contractFolderPath);
+      const verification = new Verification(
+        compilation,
+        sourcifyChainHardhat,
+        contractAddress,
+      );
+      await verification.verify();
+
+      expectVerification(verification, {
+        status: {
+          runtimeMatch: 'perfect',
+          creationMatch: null,
+        },
+        transformations: {
+          runtime: {
+            list: [
+              {
+                type: 'replace',
+                reason: 'callProtection',
+                offset: 1,
+              },
+            ],
+            values: {
+              callProtection: contractAddress.toLowerCase(),
+            },
+          },
+        },
+      });
+    });
+
+    it('should return partial match when there is perfect bytecode match but no auxdata', async () => {
+      const contractFolderPath = path.join(
+        __dirname,
+        '..',
+        'sources',
+        'NoAuxdata',
+      );
+      const { contractAddress } = await deployFromAbiAndBytecode(
+        signer,
+        contractFolderPath,
+      );
+
+      const compilation = await getCompilationFromMetadata(contractFolderPath);
+
+      const verification = new Verification(
+        compilation,
+        sourcifyChainHardhat,
+        contractAddress,
+      );
+      await verification.verify();
+
+      expectVerification(verification, {
+        status: {
+          runtimeMatch: 'partial',
+          creationMatch: null,
+        },
+      });
+    });
+
+    it('should return null match when there is no perfect match and no auxdata', async () => {
```

Comment:
I don't fully get what we are testing here? To me this is simply testing null match when bytecodes don't match?

Reply:
I changed the test's description. But it's this case: https://github.com/ethereum/sourcify/blob/de425e2bb787d6bb2457eb66f4a6fb46e466168d/packages/lib-sourcify/src/Verification/Verification.ts#L337-L340

Reply:
Ah ok got it. "no perfect match and no auxdata" was confusing to me. I made the changes to the test title and code comment to make it clear.